From 2f9ec6d840fdf9161cb02a11a98c106e263f3d64 Mon Sep 17 00:00:00 2001
From: sbruno <sbruno@FreeBSD.org>
Date: Thu, 7 Jun 2012 18:54:42 +0000
Subject: [PATCH 3/8] MFC r236323

Cosmetic nit.  If a configured volume has no label, don't emit an empty
string for the name during probe.  Simply indicate that it has no label.
---
 sys/dev/mfi/mfi_disk.c |   15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/sys/dev/mfi/mfi_disk.c b/sys/dev/mfi/mfi_disk.c
index ae5b49f..fc43e67 100644
--- a/sys/dev/mfi/mfi_disk.c
+++ b/sys/dev/mfi/mfi_disk.c
@@ -130,10 +130,17 @@ mfi_disk_attach(device_t dev)
 		state = "unknown";
 		break;
 	}
-	device_printf(dev, "%juMB (%ju sectors) RAID volume '%s' is %s\n",
-		      sectors / (1024 * 1024 / secsize), sectors,
-		      ld_info->ld_config.properties.name,
-		      state);
+
+        if ( strlen(ld_info->ld_config.properties.name) == 0 ) {
+                device_printf(dev, 
+                      "%juMB (%ju sectors) RAID volume (no label) is %s\n",
+                      sectors / (1024 * 1024 / secsize), sectors, state);
+        } else { 
+                device_printf(dev, 
+                      "%juMB (%ju sectors) RAID volume '%s' is %s\n",
+                      sectors / (1024 * 1024 / secsize), sectors,
+                      ld_info->ld_config.properties.name, state);
+        }
 
 	sc->ld_disk = disk_alloc();
 	sc->ld_disk->d_drv1 = sc;
-- 
1.7.9.5

